yt_dlp/extractor/subsplash.py

```diff
@@ -0,0 +1,186 @@
+import functools
+import json
+import math
+
+from .common import InfoExtractor
+from ..utils import (
+    InAdvancePagedList,
+    get_element_by_id,
+    int_or_none,
+    parse_iso8601,
+    try_call,
+    url_or_none,
+)
+from ..utils.traversal import traverse_obj
+
+
+class SubsplashBaseIE(InfoExtractor):
+    def _get_headers(self, url, display_id):
+        token = try_call(lambda: self._get_cookies(url)['ss-token-guest'].value)
+        if not token:
+            webpage, urlh = self._download_webpage_handle(url, display_id)
+            token = try_call(lambda: self._get_cookies(url)['ss-token-guest'].value)
+            if not token:
+                token = urlh.get_header('x-api-token')
+            if not token:
+                token = traverse_obj(get_element_by_id('shoebox-tokens', webpage), ({json.loads}, 'apiToken', {str}))
+            if not token:
+                token = self._search_regex(r'\\"tokens\\":{\\"guest\\":\\"([A-Za-z0-9._-]+)\\"', webpage, 'token', default=None)
```

Comment:
```suggestion
            token = (
                try_call(lambda: self._get_cookies(url)['ss-token-guest'].value)
                or urlh.get_header('x-api-token')
                or traverse_obj(webpage, ({find_element(id='shoebox-tokens')}, {json.loads}, 'apiToken', {str}))
                or self._search_regex(r'\\"tokens\\":{\\"guest\\":\\"([A-Za-z0-9._-]+)\\"', webpage, 'token', default=None))
```

A couple of additional points, though:

- It's usually bad practice to use the `get_element` or `find_element` utils when you need to find/parse JSON data... these utils unescape all HTML special chars which could cause JSON parsing errors. It's almost always better to use `_search_json()` instead

- If the token isn't being set as a cookie, should we cache it to a class variable?

Reply:
> It's almost always better to use `_search_json()` instead

Changed.

> If the token isn't being set as a cookie, should we cache it to a class variable?

Different tokens give different access permissions. The cookies (if present) are actually narrowly scoped by path.
I believe caching the token could lead to access denied errors, since the cached token was for a different section of the website.